src/viam/sdk/components/board.hpp

```diff
@@ -221,6 +231,22 @@ class Board : public Component {
     virtual digital_value read_digital_interrupt(const std::string& digital_interrupt_name,
                                                  const AttributeMap& extra) = 0;
 
+    /// @brief Returns a stream of digital interrupt ticks.
+    /// @param digital_interrupt_names digital interrupts to stream
+    /// @param ticks queue to put the ticks in
+    inline void stream_ticks(const std::vector<std::string> digital_interrupt_names,
```

Comment:
- The `digital_interrupt_names` parameter should probably be passed by `const&` to avoid the cost of copying it.

- I'm not clear on two things about the `ticks` parameter. First, why is it wrapped in a `shared_ptr`? Second, why is it a parameter rather than a return value? Is the idea that you should be able to repeatedly call this with the same queue to gather results periodically? If so, is there a reason it can't just be `std::queue<tick>& ticks` or `std::queue<tick>& ticks`?

- `std::queue` is a really narrow interface. I'm not sure it is helpful, in the sense that if you wanted to act on all the ticks you would need to do so destructively. I'd suggest `std::vector` as the accumulator type.

Reply:
1.) sounds good will do!

2.) The idea is that the end user will call `StreamTicks` with the queue parameter, and the ticks that occur will continuously be added to the queue and then consumed by the user. It  I tried to do `std::queue<tick>&` but when I tested with a client script the ticks never got put into the queue, making it a `shared_ptr` worked. Not clear on why that is so possible i'm implementing something wrong and don't actually need shared_ptr.

3.) I used `queue` rather than vector because most likely a user or component will be processing the ticks in the order they came in, and I don't see a super easy way to consume the first in element in a vector and then remove it. I don't see a scenario where someone would want to act on all the ticks at once, and we have the `DigitalInterruptValue` API if someone wants to see how many ticks have occurred.